### PR TITLE
[Manual Taxes M2.2] Empty State

### LINKED ISSUE #10662: [Manual Taxes M2.2] Empty state
ue2M9WUgbPdPk8Up5Sxzby-fi-365_7951

### PR DESCRIPTION
**Pull request by toupper**
Closes: #10662 

## Description

With this PR we implement the empty state for the tax rate selector:

### Changes
- Add image for the empty state to project.
- Add empty state with button below.
- Remove empty state text from the bottom notice.
- Refactor to use the same code when button actions are the same.

## Testing instructions

With a store that has no eligible tax rates (no location). The tax based on setting should be customer shipping or billing address to make the Set New Tax Rate row appear.

1. Go to Orders
2. Tap on +
3. Tap on Set New Tax Rate
4. See the empty state. Make sure that the button opens the tax settings in admin.

## Screenshots

---
- [ ] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift [hunk 1]
```diff
@@ -2,6 +2,9 @@ import SwiftUI
 import WooFoundation
 
 struct NewTaxRateSelectorView: View {
+    /// Scale of the view based on accessibility changes
+    @ScaledMetric private var scale: CGFloat = 1.0
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:6]: Nice! 💯

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift [hunk 2]
```diff
@@ -156,5 +181,10 @@ extension NewTaxRateSelectorView {
         static let taxRatesSectionTitle = NSLocalizedString("Select a tax rate", comment: "Title for the tax rate selector section")
         static let editTaxRatesInWpAdminButtonTitle = NSLocalizedString("Edit tax rates in admin",
                                                                          comment: "Title of the button that prompts the user to edit tax rates in the web")
+        static let emptyStateTitle = NSLocalizedString("We couldn’t find any tax rates", comment: "Title for the empty state on the Tax Rares selector screen")
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift]:
```suggestion
        static let emptyStateTitle = NSLocalizedString("We couldn’t find any tax rates", comment: "Title for the empty state on the Tax Rates selector screen")
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift [hunk 3]
```diff
@@ -156,5 +181,10 @@ extension NewTaxRateSelectorView {
...
+        static let emptyStateDescription = NSLocalizedString("Add tax rates in admin. Only tax rates with location information will be shown here.",
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:185]: Should we change the `admin` reference here for something like `Store Admin`, or `wp-admin`?

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift [hunk 4]
```diff
@@ -156,5 +181,10 @@ extension NewTaxRateSelectorView {
...
+                                                             comment: "Description for the empty state on the Tax Rares selector screen")
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift]:
```suggestion
                                                             comment: "Description for the empty state on the Tax Rates selector screen")
```

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift [hunk 5]
```diff
@@ -156,5 +181,10 @@ extension NewTaxRateSelectorView {
         static let taxRatesSectionTitle = NSLocalizedString("Select a tax rate", comment: "Title for the tax rate selector section")
         static let editTaxRatesInWpAdminButtonTitle = NSLocalizedString("Edit tax rates in admin",
```
Comment (iamgabrielma) [inline WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:182]: Not from this PR, but for consistency should we also rename this to `Store Admin`, or `wp-admin`?

**Review by iamgabrielma — APPROVED** [review]
:shipit:

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift (hunk 5 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:182]: This is part of the designs. In my opinion, users would know that admin refers to wp-admin, or they will discover it once they tap on it

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift (hunk 3 above)
Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:185]: Same as before

File: WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift (hunk 5 above)
Comment (iamgabrielma) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:182]: Gotcha, thanks for clarifying! We seem to use the 3 of them (admin, wp-admin, store admin) indiscriminately between features, I'll raise some examples with design to see if we can get a common approach 👍

Comment (toupper) [reply WooCommerce/Classes/ViewRelated/Orders/Order Creation/Taxes/NewTaxRateSelectorView.swift:182]: > I'll raise some examples with design to see if we can get a common approach 👍

That would be great! thanks for that

**Pull request merged**